website/source/docs/platform/k8s/ambassador.html.md

```diff
@@ -0,0 +1,271 @@
+        ---
+        apiVersion: ambassador/v1
+        kind: TLSContext
+        name: ambassador-consul
+        hosts: []
+        secret: ambassador-consul-connect
+        ---
+        apiVersion: ambassador/v1
+        kind:  Mapping
+        name:  static-service_mapping
+        prefix: /echo/
+        tls: ambassador-consul
+        service: https://static-server:443
+spec:
+    type: NodePort
+    selector:
+    app: http-echo
+    ports:
+    - port: 443
+    name: https-echo
+    targetPort: 20000            
+```
+
+Once Ambassador finishes deploying, you should have a new `LoadBalancer` service
+with a public-facing IP address. Connecting to the HTTP port on this address
+should display the output from the static service.
+
+```bash
+kubectl describe service ambassador
+
+```
+
+
+## Enabling end-to-end TLS
+
+The Ambassador service definition provided in their documentation currently does not
+serve pages over HTTPS. To enable HTTPS for full end-to-end encryption, follow
+these steps.
+
+First, upload your public SSL certificate and private key as a Kubernetes secret. 
+
+```bash
+kubectl create secret tls ambassador-certs --cert=fullchain.pem --key=privkey.pem
+```
+
+Download a copy of the [ambassador-service.yaml] file from Ambassador. Replace
+the `metadata` section with one that includes an Ambassador TLS configuration block,
+using the secret name you created in step 1. Then add an entry for port 443 to the 
+`LoadBalancer` spec. Here is a complete example:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+name: ambassador
+annotations: 
+    getambassador.io/config: |
+    ---
+    apiVersion: ambassador/v1
+    kind: Module
+    name: tls
+    config:
+    server:
+        enabled: True
+        secret: ambassador-certs
+spec:
+    type: LoadBalancer
+    externalTrafficPolicy: Local
+    ports:
+    - port: 80
+        targetPort: http
+        protocol: TCP
+        name: http
+    - port: 443
+        targetPort: https
+        protocol: TCP
+        name: https   
+    selector:
+    service: ambassador
+```
+
+Update the service definition by applying it with `kubectl`:
+
+```bash
+kubectl apply -f ambassador-service.yaml
+```
+
+You should now be able to test the SSL connection from your browser.
+
+
+## Troubleshooting
+
+When Ambassador is unable to establish an authenticated connection to the Connect proxy servers, browser connections will display this message:
+
+        upstream connect error or disconnect/reset before headers
+
+This error can have a number of different causes. Here are some things to check and troubleshooting steps you can take.
+
+### Check intentions between Ambassador and your upstream service
+
+If you followed the above installation guide, Consul should have registered a service called "ambassador". Make sure you create an intention to allow it to connect to your own services.
+
+To check whether Ambassador is allowed to connect, use the `intention check` subcommand.
```

Comment:
Maybe linking the intention check command here (https://www.consul.io/docs/commands/intention/check.html) would be useful in case I needed any of the options.

Reply:
Fixed by 9b281e09740ecf99f67f73bf270c4d37468bb909